src/test/opentrackIO_parser/python/opentrackIOlib.py

```diff
@@ -0,0 +1,304 @@
+                if (dimension == 'p'):
+                    return tr["rotation"]["pan"] * self.rot_mult
+                elif (dimension == 't'):
+                    return tr["rotation"]["tilt"] * self.rot_mult
+                elif (dimension == 'r'):
+                    return tr["rotation"]["roll"] * self.rot_mult
+                break
+
+    def Get_camera_translations(self):
+        """Return 3DOF camera coordinate: (x,y,z)"""
+        return (self.Get_trans('X'), self.Get_trans('Y'), self.Get_trans('Z'))
+
+    def Get_timecode(self):
+        """Return house timecode as a string in LTC format HH:MM:SS:FF"""
+        if ("timing" in self.pd) and (self.pd["timing"]["timecode"]):
+            hh = '{:02}'.format(int(self.pd["timing"]["timecode"]["hours"]))
+            mm = '{:02}'.format(int(self.pd["timing"]["timecode"]["minutes"]))                
+            ss = '{:02}'.format(int(self.pd["timing"]["timecode"]["seconds"]))
+            ff = '{:02}'.format(int(self.pd["timing"]["timecode"]["frames"]))
+            return hh + ':' + mm + ':' + ss + ':' + ff
+        else:
+            return None
+
+    def Get_sample_time(self,part=None):
+        """Get the PTP sample time in user-preferred format"""
+        if ("timing" in self.pd) and (self.pd["timing"]["sampleTimestamp"]):
+            ssec = int(self.pd["timing"]["sampleTimestamp"]["seconds"])
+            nsec = int(self.pd["timing"]["sampleTimestamp"]["nanoseconds"])
+            asec = int(self.pd["timing"]["sampleTimestamp"]["attoseconds"])
+            # Constants
+            epoch = 1970                        # PTP is since this epoch
+            spm = 60                            # seconds per minute. Common knowledge, but consistency is important
+            sph = 3600                          # seconds per hour
+            spd = 86400                         # sec per day
+            spy = 31536000                      # sec per year
+            # separate into years, days, hours, min, sec
+            ydelta = int(ssec / spy)            # years since epoch
+            yr = int(epoch + ydelta)            # current year
+            sty = int(ssec - (ydelta * spy))    # seconds elapsed this year
+            day = int(sty / spd)                # current day of year
+            std = int(sty - (day * spd))        # seconds elapsed today (since midnight)
+            hr = int(std / sph)                 # hours elapsed today
+            mn = int((std - (hr * sph)) / spm)  # remainder minutes
+            st = int(std - (hr * sph) - (mn * spm))  # remainder seconds
+        if not part:
+            if self.sample_time_format == "sec":
+                return ssec + (nsec * 0.000000001) + (asec * 0.000000000000000001)
+            elif self.sample_time_format == "timecode":             # since midnight
+                # FIXME: is this using the sample framerate?
+                frm = int((nsec * 0.000000001) * self.Get_timecode_framerate())
```

Comment:
I think you should use 'frame rate' here, not the timecode frame rate?

Reply:
I'd like to fix this, but I'm confused as to which part of the timing block has the relevant frame rate. Are you saying to use this frame rate?: (from the schema)
"timing": {
      "type": "object",
      "additionalProperties": false,
      "properties": {
        "frameRate"

Reply:
Done